sunpy/net/dataretriever/sources/kanzelhohe.py

```diff
@@ -0,0 +1,123 @@
+    Examples
+    --------
+    >>> from sunpy.net import Fido, attrs as a
+    >>> timerange = a.Time('2015/12/28 00:00:00','2015/12/28 09:03:00')
+    >>> results = Fido.search(timerange, a.Instrument('kanzelhohe'),
+    ...                       a.Wavelength(6563*u.AA))   #doctest: +REMOTE_DATA
+    >>> print(results)  #doctest: +REMOTE_DATA
+    Results from 1 Provider:
+    <BLANKLINE>
+    1 Results from the KanzelhoheClient:
+         Start Time           End Time      ...   Instrument      Wavelength
+           str19               str19        ...     str14           str15
+    ------------------- ------------------- ... -------------- ---------------
+    2015-12-28 00:00:00 2015-12-28 09:03:00 ... Kanzelhohe HA2 6563.0 Angstrom
+    <BLANKLINE>
+    <BLANKLINE>
+    >>> response = Fido.fetch(results)  #doctest: +SKIP
+    """
+
+    def _get_url_for_timerange(self, timerange, **kwargs):
+        """
+        Return list of URLS corresponding to value of input timerange.
+
+        Parameters
+        ----------
+        timerange: `~sunpy.time.TimeRange`
+            time range for which data is to be downloaded.
+
+        Returns
+        -------
+        urls : `list`
+            list of URLs corresponding to the requested time range
+        """
+        wave = kwargs['wavelength']
+        wave_float = (wave.min if isinstance(wave, a.Wavelength) else wave.min()).value
+        table = {6563: ['halpha2k/recent', 'halph_fr', datetime.datetime(2000, 7, 20, 7, 45, 46)],
+                 32768: ['caiia', 'caiik_fi', datetime.datetime(2000, 7, 20, 7, 45, 46)],
+                 5460: ['phokada', 'bband_fi', datetime.datetime(2000, 7, 20, 7, 45, 46)]}
+        # Checking if value is close enough to a wavelength value.
+        # Converting from one unit to other introduces precision errors.
+        wavelengths = table.keys()
+        wave = list(filter(lambda x: np.isclose(x, wave_float), wavelengths))
+        if(len(wave) == 0):    # if wave is empty
+            raise ValueError("no value is enough closer to allowed wavelengths")
+        start_date = table[wave[0]][2]
+        if timerange.start < start_date:
+            msg = 'Earliest date for which Kanzelhohe data is available is {:%Y-%m-%d}'
+            raise ValueError(msg.format(start_date))
+        prefix = "http://cesar.kso.ac.at/{datatype}/%Y/"
+        suffix = ""
+        if wave[0] != 6563:
+            suffix = "%Y%m%d/processed/"
+        url_pattern = prefix + suffix + "kanz_{datatype1}_%Y%m%d_%H%M%S.fts.gz"
+        scraper = Scraper(url_pattern, datatype=table[wave[0]][0], datatype1=table[wave[0]][1])
+        if not timerange:
+            return []
+        result = scraper.filelist(timerange)
+        return result
+
+    def _makeimap(self):
+        """
+        Helper Function:used to hold information about source.
+        """
+        self.map_['source'] = 'Global Halpha Network'  # TODO: check with kanzelhohe
```

Comment:
Probably, to keep it inline with VSO, we should use `KSO` as source and provider.